### PR TITLE
added Python script to compare sensitivity kernel images in BuildBot

### PR DESCRIPTION
**Pull request by komatits**

### CODE REVIEW

**Comment by buildbot-princeton** [comment]
The files changed require a buildbot test. Buildbot has started.

**Comment by QuLogic** [comment]
It this `rho_kappa_kernel_modified.png` really necessary?

File: EXAMPLES/LuoYang_fluid_solid_kernel/compare_two_images_to_compare_sensitivity_kernels.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+#
+# For the 2D code we can apply this to EXAMPLES/LuoYang_fluid_solid_kernel for instance in the nightly builds.
+#
+# For the 3D code for now we could just draw pictures along a couple of cut planes and use that script as well. Better than nothing.
+
+# import the necessary packages
+from skimage.measure import structural_similarity as ssim
+import matplotlib.pyplot as plt
+import numpy as np
+import cv2
```
Comment (QuLogic) [inline EXAMPLES/LuoYang_fluid_solid_kernel/compare_two_images_to_compare_sensitivity_kernels.py]: This is a pretty heavy dependency. It's also probably unnecessary. You can read PNG images using `matplotlib.image.imread`.

**Comment by buildbot-princeton** [comment]
Your changes passed the buildbot test. See [build 181](http://specfem2d-buildbot.princeton.edu/builders/specfem2d_builder/builds/181).

**Comment by QuLogic** [comment]
If this script is for BuildBot, I think it should be added to @luet's [CompareSeismo](https://github.com/luet/CompareSeismo) instead of here.

**Comment by komatits** [comment]
accept

On 04/24/2015 01:01 AM, buildbot-princeton wrote:
> Your changes passed the buildbot test. See build 181
> <http://specfem2d-buildbot.princeton.edu/builders/specfem2d_builder/builds/181>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geodynamics/specfem2d/pull/302#issuecomment-95741010>.
>

-- 
Dimitri Komatitsch
CNRS Research Director (DR CNRS), Laboratory of Mechanics and Acoustics,
UPR 7051, Marseille, France    http://komatitsch.free.fr

File: EXAMPLES/LuoYang_fluid_solid_kernel/compare_two_images_to_compare_sensitivity_kernels.py (hunk 1 above)
Comment (komatits) [inline EXAMPLES/LuoYang_fluid_solid_kernel/compare_two_images_to_compare_sensitivity_kernels.py]: not written by me, but skimage is needed (computes the SSIM index); 
opencv is used to convert to grayscale.

On 04/24/2015 12:59 AM, Elliott Sales de Andrade wrote:
> In
> EXAMPLES/LuoYang_fluid_solid_kernel/compare_two_images_to_compare_sensitivity_kernels.py
> <https://github.com/geodynamics/specfem2d/pull/302#discussion_r29011702>:
>
>> +# requires the "python-skimage" and "python-opencv" packages to be installed
>> +############################################################################
>> +
>> +# In 3D comparing kernels i.e. volumetric stuff would be a bit difficult, however in 2D we can see them as images if we draw them and then we can use Python routines that compute a similarity index between two images.
>> +#
>> +# I took one from http://www.pyimagesearch.com/2014/09/15/python-compare-two-images/ and slightly modified it. It works well. You then just need to check if the Structural Similarity Index (SSIM) is greater than 0.99.
>> +#
>> +# For the 2D code we can apply this to EXAMPLES/LuoYang_fluid_solid_kernel for instance in the nightly builds.
>> +#
>> +# For the 3D code for now we could just draw pictures along a couple of cut planes and use that script as well. Better than nothing.
>> +
>> +# import the necessary packages
>> +from skimage.measure import structural_similarity as ssim
>> +import matplotlib.pyplot as plt
>> +import numpy as np
>> +import cv2
>
> This is a pretty heavy dependency. It's also probably unnecessary. You
> can read PNG images using |matplotlib.image.imread|.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geodynamics/specfem2d/pull/302/files#r29011702>.
>

-- 
Dimitri Komatitsch
CNRS Research Director (DR CNRS), Laboratory of Mechanics and Acoustics,
UPR 7051, Marseille, France    http://komatitsch.free.fr

**Comment by komatits** [comment]
yes, until next week

On 04/24/2015 12:57 AM, Elliott Sales de Andrade wrote:
> It this |rho_kappa_kernel_modified.png| really necessary?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geodynamics/specfem2d/pull/302#issuecomment-95739898>.
>

-- 
Dimitri Komatitsch
CNRS Research Director (DR CNRS), Laboratory of Mechanics and Acoustics,
UPR 7051, Marseille, France    http://komatitsch.free.fr

**Comment by luet** [comment]
I'd rather we put all the scripts somewhere in the main repo. I should move
CompareSeismo there. In part I think it's good for developers to have
examples of scripts to submit to buildbot.

On Thu, Apr 23, 2015 at 7:01 PM, Elliott Sales de Andrade <
notifications@github.com> wrote:

> If this script is for BuildBot, I think it should be added to @luet
> <https://github.com/luet>'s CompareSeismo
> <https://github.com/luet/CompareSeismo> instead of here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geodynamics/specfem2d/pull/302#issuecomment-95741066>.
>



-- 
David Luet
Linux Administrator/Software & Programming Analyst
Department of Geosciences & PICSciE
Princeton University

**Comment by komatits** [comment]
I agree. This way people can see what is done and can also easily modify 
/ enrich them.

On 04/24/2015 02:57 PM, David Luet wrote:
> I'd rather we put all the scripts somewhere in the main repo. I should move
> CompareSeismo there. In part I think it's good for developers to have
> examples of scripts to submit to buildbot.
>
> On Thu, Apr 23, 2015 at 7:01 PM, Elliott Sales de Andrade <
> notifications@github.com> wrote:
>
>  > If this script is for BuildBot, I think it should be added to @luet
>  > <https://github.com/luet>'s CompareSeismo
>  > <https://github.com/luet/CompareSeismo> instead of here.
>  >
>  > —
>  > Reply to this email directly or view it on GitHub
>  >
> <https://github.com/geodynamics/specfem2d/pull/302#issuecomment-95741066>.
>  >
>
>
>
> --
> David Luet
> Linux Administrator/Software & Programming Analyst
> Department of Geosciences & PICSciE
> Princeton University
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geodynamics/specfem2d/pull/302#issuecomment-95924378>.
>

-- 
Dimitri Komatitsch
CNRS Research Director (DR CNRS), Laboratory of Mechanics and Acoustics,
UPR 7051, Marseille, France    http://komatitsch.free.fr

**Comment by komatits** [comment]
removed.

On 04/24/2015 01:28 AM, Dimitri Komatitsch wrote:
> yes, until next week
>
> On 04/24/2015 12:57 AM, Elliott Sales de Andrade wrote:
>> It this |rho_kappa_kernel_modified.png| really necessary?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/geodynamics/specfem2d/pull/302#issuecomment-95739898>.
>>
>>
>

-- 
Dimitri Komatitsch
CNRS Research Director (DR CNRS), Laboratory of Mechanics and Acoustics,
UPR 7051, Marseille, France    http://komatitsch.free.fr

**Pull request merged**